File: classes.py
# Define a Class
class House:

    ## self does the same job as "this"
    ## arguments are any parameter after "self", in this case with a default value
    def __init__(self, model = "a"):
        self.square_feet = 1000
        self.bathrooms = 2
        if(model == "b"):
            self.bedrooms = 2
        else:
            self.bedrooms = 3

    ## Create an inspect method
    def inspect(self):
        print(self.__dict__)

# Instantiate an instance of the class
house = House()
house2 = House("b")

print(house) ## Print the type of object and it's memory
house.inspect()## Convert to Dictionary for printing purposes

print(house2) ## Print the type of object and it's memory
house2.inspect() ## Convert to Dictionary for printing purposes

'''
# Define a Class
class House:
    
    ## self does the same job as "this"
    ## arguments are any parameter after "self", in this case with a default value
    def __init__(self, model = "a"):
        self.square_feet = 1000
        self.bathrooms = 2
        if(model == "b"):
            self.bedrooms = 2
        else:
            self.bedrooms = 3
            
    ## Create an inspect method
    def inspect(self):
        print(self.__dict__)
        
class Duplex(House):
    
    def __init__(self, model):
        super().__init__(model) ## call parent constructor
        

# Instantiate an instance of the class
house = House()
house2 = House("b")

print(house) ## Print the type of object and it's memory
house.inspect()## Convert to Dictionary for printing purposes

print(house2) ## Print the type of object and it's memory
house2.inspect() ## Convert to Dictionary for printing purposes

duplex = Duplex("b")
print(duplex)
duplex.inspect()
'''

'''
# Define a Class
class House:
    
    number_of_houses = 0 ## Class Property for the Number of Houses
    
    ## self does the same job as "this"
    ## arguments are any parameter after "self", in this case with a default value
    def __init__(self, model = "a"):
        House.addHouse()
        self.square_feet = 1000
        self.bathrooms = 2
        if(model == "b"):
            self.bedrooms = 2
        else:
            self.bedrooms = 3
            
    ## Create an inspect method
    def inspect(self):
        print(self.__dict__)
        
    ## create a class method, cls replaces self and represent the class not the instance
    @classmethod
    def addHouse(cls):
        cls.number_of_houses += 1
    
        
class Duplex(House):
    
    def __init__(self, model):
        super().__init__(model) ## call parent constructor
        

# Instantiate an instance of the class
house = House()
house2 = House("b")

print(house) ## Print the type of object and it's memory
house.inspect()## Convert to Dictionary for printing purposes

print(house2) ## Print the type of object and it's memory
house2.inspect() ## Convert to Dictionary for printing purposes

duplex = Duplex("b")
print(duplex)
duplex.inspect()

## Printing a Class Property
print(House.number_of_houses)
'''

File: classes.js
// Defining a class
class House {

    // defining an argument to the constructor function with a default value
    constructor(model = "a"){
        //"this" represent the object currently being constructed
        this.squareFeet = 1000
        this.bathrooms = 2
        if (model === "b"){
            this.bedrooms = 2
        } else {
            this.bedrooms = 3
        }
    }

/*// Create an inspect method for House objects
  inspect(){
    console.log(this)
  }*/

}


// Instantiate an instance of the class
const house = new House("a")
const house2 = new House("b")
/*
house.inspect()
house2.inspect()
*/
console.log(house)
console.log(house2)

/*
// Defining a class
class House {
  // defining an argument to the constructor function with a default value
  constructor(model = "a") {
    //"this" represent the object currently being constructed
    this.squareFeet = 1000;
    this.bathrooms = 2;
    if (model === "b") {
      this.bedrooms = 2;
    } else {
      this.bedrooms = 3;
    }
  }

  // Create an inspect method for House objects
  inspect(){
    console.log(this)
  }
}

class Duplex extends House {

    constructor(model = "a"){
        super(model) // calling the parent constructor
        this.units = 2
    }

}

// Instantiate an instance of the class
const house = new House("a");
const house2 = new House("b");

house.inspect()
house2.inspect()

const duplex = new Duplex("b")
duplex.inspect()*/


/*
// Defining a class
class House {

   // create a static property, to track the number of houses
   static numberOfHouses = 0

  // defining an argument to the constructor function with a default value
  constructor(model = "a") {
    House.addHouse() // increment the count each time a house is created
    //"this" represent the object currently being constructed
    this.squareFeet = 1000;
    this.bathrooms = 2;
    if (model === "b") {
      this.bedrooms = 2;
    } else {
      this.bedrooms = 3;
    }
  }

  // Create an inspect method for House objects
  inspect(){
    console.log(this)
  }

  static addHouse(){
    House.numberOfHouses += 1
  }
}

class Duplex extends House {

    constructor(model = "a"){
        super(model) // calling the parent constructor
        this.units = 2
    }

}

// Instantiate an instance of the class
const house = new House("a");
const house2 = new House("b");

house.inspect()
house2.inspect()

const duplex = new Duplex("b")
duplex.inspect()

console.log(House.numberOfHouses)
*/
